Data/HashMap/Base.hs

```diff
@@ -1398,6 +1402,112 @@ alterFEager f !k m = (<$> f mv) $ \fres ->
 {-# INLINABLE alterFEager #-}
 #endif
 
+-- | /O(n*m)/ Subset of maps. A map is a subset of another map if the keys are
+-- subsets and the corresponding values are equal:
+--
+-- >>> subset m1 m2 = keys m1 ⊆ keys m2 && and [ v1 == v2 | (k1,v1) <- toList m1; let v2 = m2 ! k1 ]
+--
+-- This defines a partial order on maps, for which 'union' is the least upper bound.
+-- More specifically, @subset m1 (union m1 m2)@ and @subset m2 (union m1 m2)@.
+--
+-- ==== __Examples__
+--
+-- >>> subset (fromList [(1,'a')]) (fromList [(1,'a'),(2,'b')])
+-- True
+--
+-- >>> subset (fromList [(1,'a'),(2,'b')]) (fromList [(1,'a')])
+-- False
+subset :: (Eq k, Hashable k, Eq v) => HashMap k v -> HashMap k v -> Bool
```

Comment:
In `containers`, this function is named [`isSubmapOf`](http://hackage.haskell.org/package/containers-0.6.2.1/docs/Data-Map-Strict-Internal.html#v:isSubmapOf).